js-miniapp-sdk/src/miniapp.ts

```diff
@@ -10,9 +10,23 @@ interface MiniApp {
   requestLocationPermission(): Promise<string>;
 }
 
+/**
+ * A contract declaring the interaction mechanism between mini-apps and native host app to display ads.
+ */
+interface Ad {
+  /**
+   * Requests bridge to show a Rewarded Ad.
+   * Promise is resolved with an object after the user closes the Ad.
+   * The object contains the reward earned by the user.
+   * Reward amount will be null if the user did not earn the reward.
+   * Promise is rejected if the Ad failed to load.
+   */
+  showRewardedAd(): Promise<RewardedAdResponse>;
+}
+
 /** @internal */
 /* tslint:disable:no-any */
-export class MiniAppImp implements MiniApp {
+export class MiniAppImp implements MiniApp, Ad {
```

Comment:
I looked at the generated documentation for this, and it seems kind of weird. It lists the interfaces for `MiniApp` and `Ad` but it doesn't show the combined interface anywhere, so it's not clear that all of the methods are available from a single object. 

I think we set this up slightly incorrectly. I think `MiniAppImp` needs to be named `MiniApp` and then remove the `/** @internal */` from above so that this class shows up in the docs. Then rename the `MiniApp` interface to `MiniAppFeatures` or something. basically like this:

```
interface MiniAppFeatures {
 // etc...
}

interface Ad {
  // ...
}

export class MiniApp implements MiniAppFeatures, Ad {
  // ...
}
```

What do you think?

Reply:
Yes! Generated document are splitted! Kind of hard to read!
But Typescript now reccomends not to use any suffix or prefix for interface names!
Back in the days c# used IMiniApp ~ like that interface names!!

Reply:
I think it should generate two separate interfaces, they are separate interfaces and we are only implementing them combined in a class.

Reply:
The class is flagged as @internal and the docs sh has -stripInternal option so I think the class doc is not generated! 
Removing @Internal from the class will generate document for the class combining two interfaces.
Should we remove the @internal tag from the class?

Reply:
@Abhijit-majumder Yeah, that's what I was referring to in my comment. We need to remove `@internal` so the class will show up in the documentation. And then I think the interface and class names should be renamed to what I mentioned above: `MiniApp` renamed to `MiniAppFeatures`, `MiniAppImp` renamed to `MiniApp`.